skema/text_reading/text_reading/src/main/scala/org/ml4ai/skema/text_reading/attachments/AutomatesAttachment.scala

```diff
@@ -35,7 +35,32 @@ class GroundingAttachment(candidates:Seq[GroundingCandidate]) extends AutomatesA
   })
 }
 
-class MentionLocationAttachment(filename: String, pageNum: Seq[Int], blockIdx: Seq[Int], attType: String) extends AutomatesAttachment {
+class LocationContextAttachment(locations:Seq[Mention]) extends AutomatesAttachment {
+  override def toJson: JsValue = Json.arr(locations map {
```

Comment:
There aren't any external projects tapping into our code, so we are safe on that front. Maybe we should brainstorm soon about a systematic way: For example, we could use uJson across the board and write a helper function that translates back and forth from PlayJson to uJason an arbitrary Json object which is only plugged into the controller's handlers at request and response time.

About the change, it was necessary to remove redundancy. So it was deliberate :)

Reply:
The size of the board in across the board might be an issue.  Several CLU Lab projects. including processors, have used json4s so that it is already a dependency.  In that sense ujson is the odd one out.  For play webapps there may be a way to convince it that some text is already serialized json so that it can be used from json4s or ujson without needing to be converted first to play format.  Processors has a new webapp, so I may check it out.  This ujson part of skema is still quite small, so I'm not very worried about it yet unless there are big plans ahead.  I have a gromet (de)serializer for Scala and used json4s FWIW.

Reply:
I have no special attachment to uJson. Json4S should work too. As long as we commit to one, I am happy.